### PR TITLE
Respect provided region for create cluster aws

### PR DESCRIPTION
**Pull request by ascheman**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Description

If a region is provided (`--region XXX`) there is no need to look it up
in a cluster configuration for AWS (in particular if a new cluster should
be created and there has never be a cluster configuration up to then).

#### Special notes for the reviewer(s)

Cf. https://kubernetes.slack.com/archives/C9MBGQJRH/p1573511916278100

#### Which issue this PR fixes

fixes jenkins-x/jx#6089

### CODE REVIEW

**Comment by ascheman** [comment]
Unfortunately I cannot check the failed tests. Could you enable me @dgozalo or @jstrachan ?

**Comment by hferentschik** [comment]
> Unfortunately I cannot check the failed tests. Could you enable me @dgozalo or @jstrachan ?

Unfortunately, we don't have a public view for this status checks atm. Let me see what I can do.

**Comment by hferentschik** [comment]
Regarding the failing DCO check - you will need to sign-off your commits.

See:

- https://github.com/jenkins-x/jx/pull/<pr>/checks
- https://jenkins-x.io/docs/contributing/code/#signoff

**Comment by hferentschik** [comment]
Regarding the _lint_ check. There is a Go sytle violation:

```
 // GetAccountIDAndRegion returns the current account ID and region
 func GetAccountIDAndRegion(profile string, region string) (string, string, error) {
 	sess, err := NewAwsSession(profile, region)
-	if (region == "") {
+	if region == "" {
 		// TODO for the original author (dgozalo): Is this right? We are setting up a new cluster - so we do not have a current cluster!?
 		// We need to get the region from the connected cluster instead of the one configured for the calling user
 		// as it might not be found and it would then use the default (us-west-2)
```

I recommend you run `make fmt` on your changes. That will reformat your code according to the Golang standard. You can then amend your commit with reformatted code.

**Comment by ascheman** [comment]
Thanks @hferentschik, I obviously did too much Java programming instead of GoLang lately ;-)
Great that you use/support GoLang formatting in the mean time!

File: pkg/cloud/amazon/ecr.go [hunk 1]
```diff
@@ -19,11 +19,14 @@ import (
 // GetAccountIDAndRegion returns the current account ID and region
 func GetAccountIDAndRegion(profile string, region string) (string, string, error) {
 	sess, err := NewAwsSession(profile, region)
-	// We nee to get the region from the connected cluster instead of the one configured for the calling user
-	// as it might not be found and it would then use the default (us-west-2)
-	_, region, err = GetCurrentlyConnectedRegionAndClusterName()
-	if err != nil {
-		return "", "", err
+	if (region == "") {
+		// TODO for the original author (dgozalo): Is this right? We are setting up a new cluster - so we do not have a current cluster!?
```
Comment (dgozalo) [inline pkg/cloud/amazon/ecr.go]: You are right. I think this function has been tweaked twice but only in the context of already being connected to a cluster as it was needed for things like ECR where it made sense to use the cluster region. 

It's right to use the provided region when it's not empty and use the cluster one when it is. We may have to make it more clear in the future, but for now we need this to let people create clusters.

Comment (dgozalo) [reply pkg/cloud/amazon/ecr.go]: It's also true that, if you have your region configured in your local AWS credentials / configuration it should be taken as the region to create the cluster.

**Comment by hferentschik** [comment]
@ascheman @dgozalo - just touching base on this PR. What needs to be done?

@ascheman are you planning to push an update?

**Comment by ascheman** [comment]
Sorry, didn't find the time to set up the GPG things up to now ...

**Comment by hferentschik** [comment]
> Sorry, didn't find the time to set up the GPG things up to now ...

no worries. Let us know if you need more feedback/help. Looking forward to an update of this PR.

**Comment by hferentschik** [comment]
@dgozalo @ascheman - what the status on this pull request? It needs another update, right?

**Comment by hferentschik** [comment]
@ascheman, I am closing this pull request for now. If you get around updating it feel free to open a new pull request. That would be awesome.

**Pull request closed without merging**